rasa/shared/core/training_data/structures.py

```diff
@@ -407,7 +407,11 @@ def fingerprint(self) -> Text:
         Returns:
             fingerprint of the stories
         """
-        self_as_string = self.as_story_string()
+        from rasa.shared.core.training_data.story_writer.yaml_story_writer import (
+            YAMLStoryWriter,
+        )
+
+        self_as_string = YAMLStoryWriter().dumps(self.story_steps)
```

Comment:
So this will still not be stable with `or` steps right? But we will fix on master?